p2p/ping_test.go

```diff
@@ -20,6 +20,9 @@ func TestPing(t *testing.T) {
 
 	// Convert from test nodes to non-test nodes.
 	peers := asQriNodes(testPeers)
+	for _, peer := range peers {
+		peer.Discovery.Close()
```

Comment:
One major discovery during this process: we need to ensure that peer's Discovery capabilities are turned off for certain tests, and we need to make sure we `node.GoOffline` when we finish the tests.